src/java_bytecode/java_entry_point.cpp

```diff
@@ -308,6 +308,16 @@ void java_record_outputs(
   output.add_source_location()=function.location;
 
   init_code.move_to_operands(output);
+
+  // check that there is no uncaught exception
```

Comment:
Shouldn't this be done in `goto_checkt` (in future, the same might be of interest for other languages), and is enforcing without giving any command-line control to the user a good idea?